#! /bin/sh -e
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Increases VG_N_THREADS 

[ -f debian/patches/00patch-opts ] && . debian/patches/00patch-opts
patch_opts="${patch_opts:--f --no-backup-if-mismatch}"

if [ $# -ne 1 ]; then
    echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
    exit 1
fi
case "$1" in
       -patch) patch $patch_opts -p1 < $0;;
       -unpatch) patch $patch_opts -p1 -R < $0;;
        *)
                echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
                exit 1;;
esac

exit 0
@DPATCH@

diff -urN valgrind-3.5.0.orig/include/pub_tool_threadstate.h valgrind-3.5.0/include/pub_tool_threadstate.h
--- valgrind-3.5.0.orig/include/pub_tool_threadstate.h	2009-08-19 13:37:07.000000000 +0000
+++ valgrind-3.5.0/include/pub_tool_threadstate.h	2009-09-09 15:42:44.000000000 +0000
@@ -36,7 +36,7 @@
    scheduler algorithms is surely O(N) in the number of threads, since
    that's simple, at least.  And (in practice) we hope that most
    programs do not need many threads. */
-#define VG_N_THREADS 300
+#define VG_N_THREADS 500
 
 /* Special magic value for an invalid ThreadId.  It corresponds to
    LinuxThreads using zero as the initial value for
